services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/NIHOAuth2Provider.java

```diff
@@ -0,0 +1,86 @@
+package org.sagebionetworks.repo.manager.oauth;
+
+import org.sagebionetworks.repo.model.UnauthorizedException;
+import org.sagebionetworks.repo.model.oauth.ProvidedUserInfo;
+import org.sagebionetworks.repo.model.principal.AliasType;
+import org.sagebionetworks.repo.web.OAuthException;
+import org.scribe.model.OAuthConfig;
+import org.scribe.model.OAuthRequest;
+import org.scribe.model.Response;
+import org.scribe.model.Token;
+import org.scribe.model.Verb;
+import org.scribe.model.Verifier;
+import org.scribe.oauth.OAuthService;
+
+import static org.sagebionetworks.repo.manager.oauth.GoogleOAuth2Provider.parserResponseBody;
```

Comment:
NIHOAuth2Provider should not be referring to GoogleOAuth2Provider.  If `parserResponseBody()` is the same for GoogleOAuth2Provider and NIHOAuth2Provider, then factor it out into a helper class.